### PR TITLE
[RT-434] Synapse api client

### PR DESCRIPTION
**Pull request by sinfantes4**
### What is the goal?
Create gem skeleton for synapse Api Client

### References
* **Issue:** https://sequra.atlassian.net/browse/RT-434

### How is it being implemented?
Added Dockerfile-development, docker-compose file and .env
Created new gem. Configure gemspec
Add sequra-style gem

### How is it tested?
Automatic tests. Created a test for the version

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+SQ_RUBY_VERSION=3.2.2
+SQ_ALPINE_VERSION=3.17
```
Comment (rpr8895) [inline .env:2]: Is this not a development file?

Comment (sinfantes4) [reply .env:2]: mm, what do you mean? you mean that is should be git ignored?

File: Gemfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+source "https://rubygems.org"
+
+# Specify your gem's dependencies in synapse-api-client.gemspec
+gemspec
+
```
Comment (albertvinyes) [inline Gemfile:7]: I have one question, why don't we add `ruby "3.2.2"` in the Gemfile?

File: lib/synapse/version.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
+
+module Synapse
+  VERSION = "0.1.0"
```
Comment (albertvinyes) [inline lib/synapse/version.rb:4]: Not sure if we can benefit from github actions here

File: LICENSE.txt [hunk 4]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2023 TODO: Write your name
```
Comment (albertvinyes) [inline LICENSE.txt:3]: I guess we want to specify the organization or your name here

Comment (sinfantes4) [reply LICENSE.txt:3]: 😅

File: .env (hunk 1 above)
Comment (sinfantes4) [reply .env:2]: It is used to define environment variables. In this case, yes, only for development, like the docker and docker-compose files. But we need it in the repo. docker-compose file uses it. If you clone the repo you need those variables defined somewhere

File: lib/synapse/version.rb (hunk 3 above)
Comment (sinfantes4) [reply lib/synapse/version.rb:4]: Yes, actually this file will be changed and maintained by github actions

**Pull request merged**